src/ComponentInstaller.php

```diff
@@ -535,18 +539,44 @@ private function removePackageFromConfig($package, array $metadata, Collection $
      */
     private function removeModuleFromConfig($module, $package, Collection $injectors)
     {
-        $injectors->each(function ($injector) use ($module, $package) {
+        $injectors->each(function (InjectorInterface $injector) use ($module, $package) {
             $this->io->write(sprintf('<info>Removing %s from package %s</info>', $module, $package));
 
             if ($injector->remove($module)) {
                 $this->io->write(sprintf(
                     '<info>    Removed package from %s</info>',
-                    $injector->getConfigFile()
+                    $this->getInjectorConfigFileName($injector)
                 ));
             }
         });
     }
 
+    /**
+     * @param InjectorInterface $injector
+     * @return string
+     */
+    private function getInjectorConfigFileName(InjectorInterface $injector)
+    {
+        $fileName = '';
+        if ($injector instanceof ConfigInjectorChain) {
+            // obtain all included names from injectors chain
+            $chain = $injector->getCollection();
+            $names = [];
+            foreach ($chain as $item) {
+                $names[] = $this->getInjectorConfigFileName($item);
+            }
+            $fileName = implode(', ', $names);
+        } elseif ($injector instanceof NoopInjector) {
+            // virtual file name for stub
+            $fileName = 'no-op.conf';
+        } elseif ($injector instanceof AbstractInjector) {
+            // default flow
+            $fileName = $injector->getConfigFile();
+        }
```

Comment:
Please split each of these cases into its own method, and return the results of calling that method.

As an example:

```php
private function getInjectorConfigFileName(InjectorInterface $injector)
{
    if ($injector instanceof ConfigInjectorChain) {
        return $this->getConfigFileNameFromInjectorChain($injector);
    }
    // etc.
    return '';
}

private function getConfigFileNameFromInjectorChain(ConfigInjectorChain $injector)
{
    $names = [];
    foreach ($injector->getCollection() as $item) {
        $names[] = $this->getInjectorConfigFileName($item);
    }
    return implode(', ', $names);
}
```

Reply:
@weierophinney done; removed the noop case as it's not actually possible to reach it with current ConfigDiscovery logic